lib/src/widgets/yaru_search_field.dart

```diff
@@ -0,0 +1,372 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:yaru/yaru.dart';
+import 'package:yaru_icons/yaru_icons.dart';
+import 'package:yaru_widgets/constants.dart';
+import 'package:yaru_widgets/src/widgets/yaru_icon_button.dart';
+
+/// A [TextField] to with fully rounded corners,
+/// ideally in a [YaruWindowTitleBar] or [YaruDialogTitleBar]
+class YaruSearchField extends StatefulWidget {
+  const YaruSearchField({
+    super.key,
+    this.text,
+    this.onSubmitted,
+    this.hintText,
+    this.height = kYaruTitleBarItemHeight,
+    this.contentPadding = const EdgeInsets.only(
+      bottom: 10,
+      top: 10,
+      right: 15,
+      left: 15,
+    ),
+    this.autofocus = true,
+    this.onClear,
+    this.onChanged,
+    this.radius = const Radius.circular(kYaruTitleBarItemHeight),
+    this.style = YaruSearchFieldStyle.filled,
+    this.borderColor,
+    this.fillColor,
+    this.controller,
+    this.focusNode,
+  });
+
+  /// Optional [String] forwarded to the internal [TextEditingController]
+  final String? text;
+
+  /// Optional [String] used inside the internal [InputDecoration]
+  final String? hintText;
+
+  /// The callback forwarded to the [TextField] used when the enter key is pressed
+  final void Function(String? value)? onSubmitted;
+
+  /// The callback forwarded to the [TextField] used when input changes
+  final void Function(String value)? onChanged;
+
+  /// Optional callback used to clear the [TextField]. If provided an [IconButton] will use it
+  /// as the suffix icon inside the [InputDecoration]
+  final void Function()? onClear;
+
+  /// The height of the [TextField] that defaults to [kYaruTitleBarItemHeight]
+  final double height;
+
+  /// The padding for the [InputDecoration] that defaults to `EdgeInsets.only(bottom: 10,top: 10, right: 15, left: 15)`
+  final EdgeInsets contentPadding;
+
+  /// Defines if the [TextField] is autofocused on build
+  final bool autofocus;
+
+  /// Defines the radius for the corners.
+  final Radius radius;
+
+  final YaruSearchFieldStyle style;
+
+  /// Optional [Color] for the border. If not provided
+  /// it will fall back to `ColorScheme.dividerColor`.
+  final Color? borderColor;
+
+  /// Optional [Color] for the border. If not provided
+  /// it will fall back to `ColorScheme.dividerColor`.
+  final Color? fillColor;
+
+  /// Optional controller for the internal [TextField]
+  final TextEditingController? controller;
+
+  /// Optional [FocusNode] for the internal [KeyboardListener]
+  final FocusNode? focusNode;
+
+  @override
+  State<YaruSearchField> createState() => _YaruSearchFieldState();
+}
+
+class _YaruSearchFieldState extends State<YaruSearchField> {
+  late TextEditingController? _controller;
+  late FocusNode? _focusNode;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = widget.controller ?? TextEditingController(text: widget.text);
+    _focusNode = widget.focusNode;
+  }
+
```

Comment:
the usual rebuild checks, something like
```dart
  @override
  void didUpdateWidget(YaruSearchField oldWidget) {
    super.didUpdateWidget(oldWidget);

    if (widget.controller != oldWidget.controller) {
      if (oldWidget.controller == null) _controller.dispose();
      _controller = widget.controller ?? TextEditingController(text: widget.text);
    }
  }
```

Reply:
thanks, also for focusnode?

Reply:
oh yes that too :+1:

Reply:
okay did this